File: jcplayer/src/main/java/com/example/jean/jcplayer/service/notification/JcPlayerNotificationReceiver.java
package com.example.jean.jcplayer.service.notification;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.example.jean.jcplayer.general.errors.AudioListNullPointerException;
import com.example.jean.jcplayer.JcPlayerManager;

public class JcPlayerNotificationReceiver extends BroadcastReceiver {
    public JcPlayerNotificationReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        JcPlayerManager jcPlayerManager = JcPlayerManager.getInstance(context, null, null);
        String action = "";

        if (intent.hasExtra(JcNotificationService.ACTION)) {
            action = intent.getStringExtra(JcNotificationService.ACTION);
        }

        switch (action) {
            case JcNotificationService.PLAY:
                try {
                    jcPlayerManager.continueAudio();
                    jcPlayerManager.updateNotification();
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;

            case JcNotificationService.PAUSE:
                try {
                    if(jcPlayerManager != null) {
                        jcPlayerManager.pauseAudio();
                        jcPlayerManager.updateNotification();
                    }
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;

            case JcNotificationService.NEXT:
                try {
                    jcPlayerManager.nextAudio();
                } catch (AudioListNullPointerException e) {
                    try {
                        jcPlayerManager.continueAudio();
                    } catch (AudioListNullPointerException e1) {
                        e1.printStackTrace();
                    }
                }
                break;

            case JcNotificationService.PREVIOUS:
                try {
                    jcPlayerManager.previousAudio();
                } catch (Exception e) {
                    try {
                        jcPlayerManager.continueAudio();
                    } catch (AudioListNullPointerException e1) {
                        e1.printStackTrace();
                    }
                }
                break;
        }
    }
}

File: jcplayer/src/main/java/com/example/jean/jcplayer/service/notification/JcPlayerNotificationReceiver.kt
package com.example.jean.jcplayer.service.notification

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.example.jean.jcplayer.JcPlayerManager
import com.example.jean.jcplayer.general.errors.AudioListNullPointerException

class JcPlayerNotificationReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val jcPlayerManager = JcPlayerManager.getInstance(context)
        var action = ""

        if (intent.hasExtra(JcNotificationPlayer.ACTION)) {
            action = intent.getStringExtra(JcNotificationPlayer.ACTION)
        }

        when (action) {
            JcNotificationPlayer.PLAY -> try {
                jcPlayerManager.get()?.continueAudio()
                jcPlayerManager.get()?.updateNotification()
            } catch (e: Exception) {
                e.printStackTrace()
            }

            JcNotificationPlayer.PAUSE -> try {
                jcPlayerManager.get()?.pauseAudio()
                jcPlayerManager.get()?.updateNotification()
            } catch (e: Exception) {
                e.printStackTrace()
            }

            JcNotificationPlayer.NEXT -> try {
                jcPlayerManager.get()?.nextAudio()
            } catch (e: AudioListNullPointerException) {
                try {
                    jcPlayerManager.get()?.continueAudio()
                } catch (e1: AudioListNullPointerException) {
                    e1.printStackTrace()
                }

            }

            JcNotificationPlayer.PREVIOUS -> try {
                jcPlayerManager.get()?.previousAudio()
            } catch (e: Exception) {
                try {
                    jcPlayerManager.get()?.continueAudio()
                } catch (e1: AudioListNullPointerException) {
                    e1.printStackTrace()
                }
            }

        }
    }
}
